controlplane/src/core/workers/AIGraphReadmeWorker.ts

```diff
@@ -0,0 +1,112 @@
+import { PostgresJsDatabase } from 'drizzle-orm/postgres-js';
+import { ConnectionOptions, Job, Worker, Queue } from 'bullmq';
+import pino from 'pino';
+import * as schema from '../../db/schema.js';
+import { OpenAIGraphql } from '../openai-graphql/index.js';
+import { SubgraphRepository } from '../repositories/SubgraphRepository.js';
+import { FederatedGraphRepository } from '../repositories/FederatedGraphRepository.js';
+
+const QueueName = 'ai.graph-readme-generator';
+
+export interface CreateReadmeInputEvent {
+  targetId: string;
+  organizationId: string;
+  type: 'subgraph' | 'federated_graph';
+}
+
+export class AIGraphReadmeQueue {
+  private readonly queue: Queue<CreateReadmeInputEvent>;
+
+  constructor(conn: ConnectionOptions) {
+    this.queue = new Queue<CreateReadmeInputEvent>(QueueName, {
+      connection: conn,
+      defaultJobOptions: {
+        removeOnComplete: true,
+        removeOnFail: true,
+        attempts: 3,
+        backoff: {
+          type: 'exponential',
+          delay: 5000,
+        },
+      },
+    });
+  }
+
+  public async addJob(job: CreateReadmeInputEvent) {
+    await this.queue.add(`targets/${job.targetId}`, job);
+  }
+}
+
+class AIGraphReadmeWorker {
+  private readonly oaig: OpenAIGraphql;
+
+  constructor(
+    private input: {
+      redisConnection: ConnectionOptions;
+      db: PostgresJsDatabase<typeof schema>;
+      log: pino.Logger;
+      openAiApiKey: string;
+    },
+  ) {
+    this.oaig = new OpenAIGraphql({
+      openAiApiKey: input.openAiApiKey,
+    });
+    this.input.log = input.log.child({ worker: this.constructor.name });
+  }
+
+  public async handler(job: Job<CreateReadmeInputEvent>) {
+    if (job.data.type === 'subgraph') {
```

Comment:
use switch + move each job to another func?

Reply:
I've moved both into separate functions. I don't see a value here to use switch though.